src/main/scripts/ctl/owsFunctions.xml

```diff
@@ -2852,20 +2852,38 @@
   <ctl:function name="owsFunctions:check.GetCapabilities.Response.OnlineResource">
     <ctl:param name="capabilities.elt"/>
     <ctl:param name="method">get|post</ctl:param>
-    <ctl:description>Verify that each OnlineResource URL is either a URL prefix termined by "?" for HTTP Get or a URL not terminated y "?" for HTTP Post</ctl:description>
+    <ctl:description>Verify that each OnlineResource URL is either a URL prefix terminated by "?" or "&amp;" for HTTP GET, a URL for RESTful HTTP GET or a URL not terminated by "?" for HTTP POST</ctl:description>
     <ctl:code>
       <xsl:variable name="results">
         <xsl:choose>
           <xsl:when test="lower-case($method) = 'get' ">
             <xsl:for-each select="$capabilities.elt/ows:OperationsMetadata/ows:Operation/ows:DCP/ows:HTTP/ows:Get/@xlink:href">
               <xsl:variable name="urlPrefix" select="."/>
+              <xsl:variable name="operationsMetadata" select="../../../../.."/>
               <xsl:choose>
-                <xsl:when test="((starts-with($urlPrefix,'http://') or starts-with($urlPrefix,'https://')) and (ends-with($urlPrefix, '?') or ends-with($urlPrefix, '&amp;')))">
-                  <xsl:text>true|</xsl:text>
+                <xsl:when test="$operationsMetadata/ows:Operation/ows:DCP/ows:HTTP/ows:Get/ows:Constraint[@name='GetEncoding']/ows:AllowedValues/ows:Value = 'RESTful' or
```

Comment:
My tests showed, that this when-clause is always true, as long as one of the ows:Get-elements contained a value 'RESTful'. The expression `$operationsMetadata/ows:Operation/ows:DCP/ows:HTTP/ows:Get/ows:Constraint[@name='GetEncoding']/ows:AllowedValues/ows:Value` seems to return a list of values. This means, that the check for KVP-URLs is never executed. So I was able to use a malformed KVP URL (i.e. without a ? or & at the end) and the tests passed.
I have made a small modification of the code that fixes this. However, I had to remove the checks for `$operationsMetadata/ows:Operation/ows:Constraint[@name='GetEncoding']/ows:AllowedValues/ows:Value = 'RESTful'` and `$operationsMetadata/ows:Constraint[@name='GetEncoding']/ows:AllowedValues/ows:Value = 'RESTful'`.
While those two constructs are supported by the schema, there would be no URL (as this comes from the ows:Get-element, see line 2860). So the test would fail anyway. Therefore, I think it would be ok to remove those checks and only check for `$operationsMetadata/ows:Operation/ows:DCP/ows:HTTP/ows:Get/ows:Constraint[@name='GetEncoding']/ows:AllowedValues/ows:Value = 'RESTful'`.